extmod/network_wiznet5k.c

```diff
@@ -388,6 +379,17 @@ STATIC void wiznet5k_init(void) {
     };
     wiznet5k_obj.netinfo = netinfo;
 
+    // Configure wiznet provided TCP / socket interface
+
+    reg_dhcp_cbfunc(wiz_dhcp_assign, wiz_dhcp_update, wiz_dhcp_conflict);
+
+    uint8_t sn_size[16] = {2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2};         // 2k buffer for each socket
+    ctlwizchip(CW_INIT_WIZCHIP, sn_size);
+
+    ctlnetwork(CN_SET_NETINFO, (void *)&wiznet5k_obj.netinfo);
```

Comment:
This change makes sense (calling ctlnetwork after actually initialising netinfo), but maybe only the ctlnetwork line needs to move?

Reply:
Right,
In conclusion, after initializing netinfo, call the ctlnetwork function and set the network-related register to the initialized value. Before I change the code , I tried to test it  and I saw it that regs() function call through Thonny and Wiz CREG value hasn't initialized value.

Reply:
Looking into this some more...

The default behavior just about everywhere else in MicroPython (including WizNet-with-LWIP) is to not have a default address, and enable DHCP by default. So while I agree that this code is wrong as it is, I think we should just remove it altogether.